diff --git a/scripts/dtc/pylibfdt/Makefile b/scripts/dtc/pylibfdt/Makefile
index 01d5e0ffe3..2b2dfe0bd9 100644
--- a/scripts/dtc/pylibfdt/Makefile
+++ b/scripts/dtc/pylibfdt/Makefile
@@ -15,6 +15,8 @@ PYLIBFDT_srcs = $(addprefix $(LIBFDT_srcdir)/,$(LIBFDT_SRCS)) \
 
 quiet_cmd_pymod = PYMOD   $@
       cmd_pymod = unset CC; unset CROSS_COMPILE; unset CFLAGS;\
+		CC="$(HOSTCC)" \
+		LDSHARED="$(HOSTLDSHARED)" \
 		LDFLAGS="$(HOSTLDFLAGS)" \
 		VERSION="u-boot-$(UBOOTVERSION)" \
 		CPPFLAGS="$(HOSTCFLAGS) -I$(LIBFDT_srcdir)" OBJDIR=$(obj) \
-- 
2.11.0

